### PR TITLE
introduced generators/shuffle : tool to shuffle selected text

### PR DESCRIPTION
**Pull request by Shankhanil**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "jasmine": "^3.5.0"
   },
   "scripts": {
-    "test": "./node_modules/.bin/eslint ./src/main.js ./src/* ./spec/*.js && ./node_modules/.bin/jasmine --random=false",
+    "test": "eslint --fix ./src/main.js ./src/* ./spec/*.js && eslint ./src/main.js ./src/* ./spec/*.js && jasmine --random=false",
```
Comment (nadchif) [inline package.json:22]: I think it's better to keep the "--fix" argument that already exists on the `lintfix`command. You can access it by `npm run-script lintfix`

Comment (Shankhanil) [reply package.json:22]: Okay, let me try it out!

**Comment by Shankhanil** [comment]
you see, the build if failing if I don't lintfix in TravisCI. I lintfixed in my local

File: src/generators/shuffle.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ *  File: PHPSerialize.js
+ *  Author: Shankhanil <shankha.rik@gmail.com>
+ *  Description:  Shuffles a string
+ */
+
+define((require, exports) => {
+
```
Comment (nadchif) [inline src/generators/shuffle.js:8]: Block must not be padded by blank lines

File: src/generators/shuffle.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  //    PHP Serializing string
+  const shuffle = function shuffle(input) {
+    let chars = input.split(""), strlen = chars.length;
```
Comment (nadchif) [inline src/generators/shuffle.js:11]: Split 'let' declarations into multiple statements

Comment (nadchif) [reply src/generators/shuffle.js:11]: 'chars' is never reassigned. Use 'const' instead

Comment (nadchif) [reply src/generators/shuffle.js:11]: Strings must use singlequote

File: src/generators/shuffle.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    for(var i = strlen - 1; i > 0; i =i-1) {
```
Comment (nadchif) [inline src/generators/shuffle.js:13]: Expected space(s) after "for"

Comment (nadchif) [reply src/generators/shuffle.js:13]: All 'var' declarations must be at the top of the function scope

Comment (nadchif) [reply src/generators/shuffle.js:13]: Operator '=' must be spaced

File: src/generators/shuffle.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        var j = Math.floor(Math.random() * (i + 1));
```
Comment (nadchif) [inline src/generators/shuffle.js:14]: All 'var' declarations must be at the top of the function scope

File: src/generators/shuffle.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        var tmp = chars[i];
+        chars[i] = chars[j];
+        chars[j] = tmp;
+    }
+    return chars.join("");
```
Comment (nadchif) [inline src/generators/shuffle.js:19]: Strings must use singlequote

**Comment by Shankhanil** [comment]
build is passing. YAAAY!

**Comment by Shankhanil** [comment]
So is it good to be merged?

**Pull request merged**